### PR TITLE
equal frontend function

### PR DESCRIPTION
**Pull request by mustafaa7med**
equal #15849

### CODE REVIEW

**Comment by mustafaa7med** [comment]
Hello @Dsantra92,

I've looked into the 2 errors and i assume it is unrelated to the function, I assumed it is an import fail ```statistical_dtype_values```
Kindly advice

Thanks in advance,
Moustafa

**Comment by mustafaa7med** [comment]
Hello @paulaehab,

I would be grateful if you could review my PR.

Thanks in advance,
Moustafa

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -380,6 +380,12 @@ def not_equal(self, other, *, out=None):
 
     ne = not_equal
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16", "bfloat16")}, "torch")
+    def equal(self, other, *, out=None):
+        return torch_frontend.equal(self, other, out=out)
+
+    eq = equal
```
Comment (paulaehab) [inline ivy/functional/frontends/torch/tensor.py:387]: you don't need to add this as it must be added in separated issue

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -380,6 +380,12 @@ def not_equal(self, other, *, out=None):
 
     ne = not_equal
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16", "bfloat16")}, "torch")
+    def equal(self, other, *, out=None):
```
Comment (paulaehab) [inline ivy/functional/frontends/torch/tensor.py:384]: equal function don't take this arguments 😄  please check the torch docs and change it after that be sure that all tests is passing

**Review by paulaehab — CHANGES_REQUESTED** [review]

**Review by paulaehab — CHANGES_REQUESTED** [review]
after you correct changes and all tests pass I will merge it

**Comment by mustafaa7med** [comment]
Hello @paulaehab,

Thank you for your input and help, I reviewed the parameters required for this function and adjusted them accordingly, Moreover, Thanks for the referencing, It got slipped from my mind.

Unfortunately, the PR keeps failing test(1) and I'm not entirely sure what seems to be the reason, Could you please advise?
https://github.com/unifyai/ivy/actions/runs/5088963429/jobs/9146020121?pr=15852#step:3:160

Thanks in advance,
Moustafa Ahmed

**Comment by mustafaa7med** [comment]
@paulaehab Any advise?

**Comment by paulaehab** [comment]
@mustafaa7med  Hello, the problem was because ivy.equal is elementwise but torch.equal is single output
so reduce all reduces it down to single True if every entry is True else False it solved by adding ivy.all around ivy.equal around ivy.equal in comparison.py I already add it for you and Ishtiaq help us here in the PR. 
I will merge it nice work 😄 .

**Pull request merged**